### ISSUE TITLE
write Coder binary to a fixed (not-user-specific) path

### ISSUE THREAD

**Issue by spikecurtis** [coder/vscode-coder#256]
In order to get direct network connections working, a customer needs to add a firewall exception for `coder.exe` on Windows.  They manage their firewall rules centrally, and the problem is that the IDE plugin writes to the directory

```
c:\Users\%USERNAME%\AppData\Roaming\Code\User\globalStorage\coder.coder-remote
```

Since the directory is different for every person, they can't push out a central firewall rule.

It would be better if we could write to somewhere that doesn't depend on the username, and/or write somewhere centrally configurable.

**Comment by code-asher** [comment]
If they can control the settings they can configure `coder.binaryDestination` to a static location.

Not sure how easy it is to manage VS Code settings as policy, though, so maybe we need a different way to configure this.

**Comment by code-asher** [comment]
Oh yeah actually VS Code does have policy-level settings, maybe this will work: https://code.visualstudio.com/docs/setup/enterprise

**Comment by EhabY** [comment]
It seems the policy-level settings they have are a bit broad and generic. Here's the full list:

<img width="2504" height="912" alt="Image" src="https://github.com/user-attachments/assets/c9164570-45c5-47bd-b05c-bb6967331f35" />

I see a few solutions here for this to work the firewall:
* As @code-asher suggested, users have to set their `coder.binaryDestination` (which they can share by sharing `settings.json` if the path is the same)
* We can provide an environment variable for that so "it just works"
* By default the binary is installed in `<global storage path>/coder.coder-remote/<coder url>` , where the executable lives in `bin` directory. Potentially the firewall can allow such a path but it'll be different on all systems because of the username and the OS

**Comment by EhabY** [comment]
I am not very familiar with Windows firewall rules but I mentioned this as a possibility if they can have wildcards in there.

> It would be better if we could write to somewhere that doesn't depend on the username, and/or write somewhere centrally configurable.

What Asher mentioned should be that exactly then. Though sadly it cannot be centrally managed unless they write to the `settings.json` for each user. (Even then it would still be configurable by the user though it would just be blocked by the firewall).

**Comment by spikecurtis** [comment]
> One last thing, in all other settings, environment variables take lower priority than actual settings in VS Code (or CLI), should it be the same here? If it is then it would just mean that users can change the binary location but the firewall rules would just kick in and block it in that case.

Yeah, that's fine. If the user screws it up it fails secure (deny).
